TwitchPlaysAssembly/Src/ComponentSolvers/Modded/Misc/ParliamentComponentSolver.cs

```diff
@@ -0,0 +1,494 @@
+							yield return Click(0, .2f);
+						else
+							yield return Click(2, .2f);
+					}
+					else if (letterPosition >= 10 && letterPosition < 15)
+					{
+						if (serialNumber.Contains("V") || serialNumber.Contains("O") || serialNumber.Contains("T") || serialNumber.Contains("E"))
+							yield return Click(0, .2f);
+						else
+							yield return Click(2, .2f);
+					}
+					else if (letterPosition >= 15 && letterPosition < 20)
+						yield return Click(2, .2f);
+					else
+					{
+						if (opposedLastBill)
+							yield return Click(0, .2f);
+						else
+							yield return Click(2, .2f);
+					}
+				}
+			}
+		}
+		if (_component.GetValue<bool>("timeToResign") && !_component.GetValue<bool>("finalStage"))
+		{
+			int numberOfVowels = _component.GetValue<int>("numberOfVowels");
+			int numberOfConsonants = _component.GetValue<int>("numberOfConsonants");
+			int litIndicators = Module.BombComponent.GetComponent<KMBombInfo>().GetOnIndicators().Count();
+			Party party = _component.GetValue<Party>("party");
+			// Copy-pasted press handling code for the fptp and mmp button from Parliament
+			if (numberOfVowels >= (numberOfConsonants - 2))
+			{
+				if (litIndicators % 2 != 0)
+				{
+					if ((party == Party.socialist || party == Party.lan || party == Party.birthday || party == Party.conservative))
+						yield return Click(3, .2f);
+					else
+						yield return Click(1, .2f);
+				}
+				else
+				{
+					if (party == Party.liberal || party == Party.communist || party == Party.socialist)
+						yield return Click(1, .2f);
+					else
+						yield return Click(3, .2f);
+				}
+			}
+			else
+			{
+				if (litIndicators % 2 == 0)
+					yield return Click(3, .2f);
+				else
+					yield return Click(1, .2f);
+			}
+		}
+		if (_component.GetValue<bool>("finalStage"))
+		{
+			int currentPoll = _component.GetValue<int>("currentPoll");
+			int electionMethod = _component.GetValue<int>("electionMethod");
+			int numberOfBatteries = Module.BombComponent.GetComponent<KMBombInfo>().GetBatteryCount();
+			int numberOfPorts = Module.BombComponent.GetComponent<KMBombInfo>().GetPortCount();
+			Party party = _component.GetValue<Party>("party");
+			// Copy-pasted press handling code for the win and lose button from Parliament
+			if (electionMethod == 0)
+			{
+				if (((numberOfBatteries + numberOfPorts) * 10) > currentPoll)
+					yield return Click(1, .2f);
+				else
+					yield return Click(3, .2f);
+			}
+			else
+			{
+				if (currentPoll >= 40 && !(party == Party.birthday || party == Party.lan))
+					yield return Click(1, .2f);
+				else
+					yield return Click(3, .2f);
+			}
+		}
+	}
+
+	enum Party { republican, democratic, conservative, liberal, socialist, communist, birthday, lan };
+	enum BillOpener { prevent, promote, fund, endorse, condemn, oppose };
+	enum BillMiddle { healthcare, support, hats, freedom, vaccines, rights };
+	enum BillEnding { veterans, children, dogs, cats, waterfowl, liberals };
```

Comment:
Also need to change the enum order.
```suggestion
	enum Party { republican, democratic, conservative, liberal, socialist, communist, birthday, lan };
	enum BillOpener { prevent, promote, fund, endorse, oppose, condemn };
	enum BillMiddle { healthcare, support, hats, freedom, vaccines, rights };
	enum BillEnding { veterans, children, dogs, waterfowl, liberals, cats };
```